nvflare/app_common/workflows/hierarchical_statistics_controller.py

```diff
@@ -0,0 +1,288 @@
+                        "Locations": [
+                            {
+                            "Name": "Location-1",
+                            "Devices": ["Device-1", "Device-2"]
+                            },
+                            {
+                            "Name": "Location-2",
+                            "Devices": ["Device-3"]
+                            }
+                        ]
+                        },
+                        {
+                        "Name": "Org-2",
+                        "Locations": [
+                            {
+                            "Name": "Location-1",
+                            "Devices": ["Device-4", "Device-5"]
+                            },
+                            {
+                            "Name": "Location-2",
+                            "Devices": ["Device-6"]
+                            }
+                        ]
+                        }
+                    ]
+                    },
+                    {
+                    "Name": "Manufacturer-2",
+                    "Orgs": [
+                        {
+                        "Name": "Org-3",
+                        "Locations": [
+                            {
+                            "Name": "Location-1",
+                            "Devices": ["Device-7", "Device-8"]
+                            },
+                            {
+                            "Name": "Location-6",
+                            "Devices": ["Device-9"]
+                            }
+                        ]
+                        }
+                    ]
+                    }
+                ]
+            }
+
+        """
+        super().__init__(
+            statistic_configs,
+            writer_id,
+            wait_time_after_min_received,
+            result_wait_timeout,
+            precision,
+            min_clients,
+            enable_pre_run_task,
+        )
+        self.hierarchy_config = hierarchy_config
+
+    def statistics_task_flow(self, abort_signal: Signal, fl_ctx: FLContext, statistic_task: str):
+
+        self.log_info(fl_ctx, f"start prepare inputs for task {statistic_task}")
+        inputs = self._prepare_inputs(statistic_task)
+        results_cb_fn = self._get_result_cb(statistic_task)
+
+        self.log_info(fl_ctx, f"task: {self.task_name} statistics_flow for {statistic_task} started.")
+
+        if abort_signal.triggered:
+            return False
+
+        task_props = {StC.STATISTICS_TASK_KEY: statistic_task}
+        task = Task(name=self.task_name, data=inputs, result_received_cb=results_cb_fn, props=task_props)
+
+        self.broadcast_and_wait(
+            task=task,
+            targets=None,
+            min_responses=self.min_clients,
+            fl_ctx=fl_ctx,
+            wait_time_after_min_received=self.wait_time_after_min_received,
+            abort_signal=abort_signal,
+        )
+
+        if self.hierarchy_config:
```

Comment:
Since there could be errors in config, we should try to catch errors as soon as possible.
Please consider moving this block of code to start_controller(), which happens before the workflow is started.

Reply:
Done, I have added required exception handling and also moved this block to the start of the function. thanks.